### PR TITLE
[ENH]check for algorithm_type

### LINKED ISSUE #1952: [ENH] Add test that the estimator "algorithm_type" is valid
### Describe the feature or idea you want to propose

for collection estimators we use the 
    _tags = {"algorithm_type": "distance"}
as a taxonomy of algorithms. However we dont test they are valid, as identified in #1935 

### Describe your proposed solution

add a test the type is one of a valid list, probably defined in the base class?

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by aryanpola**
#### Reference Issues/PRs
Fixes #1952 

#### What does this implement/fix? Explain your changes.
Adds a check for `algorithm_type` in `aeon.testing.estimator_checking._yield_classification_checks.py`

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

**Comment by aryanpola** [comment]
@MatthewMiddlehurst should I raise errors for the absence of the `algorithm_type` tag in the estimator?

**Comment by TonyBagnall** [comment]
> @MatthewMiddlehurst should I raise errors for the absence of the `algorithm_type` tag in the estimator?

no, they do not have to have them, but if a classifier does have this tag, it should be one of 
{"convolution", "dictionary","distance", "feature", "shapelet", "interval","hybrid","deeplearning"}

thats all you need to check, thanks for your contributions

**Comment by aryanpola** [comment]
@TonyBagnall I've added `None` to the list of valid algorithms since some have it as none

File: aeon/testing/estimator_checking/_yield_classification_checks.py [hunk 1]
```diff
@@ -126,6 +133,32 @@ def check_classifier_against_expected_results(estimator_class):
...
+        "dictionary",
+        "interval",
+        "feature",
+        "hybrid",
+        "shapelet",
+        "None",
+    ]
+    algorithm_type = estimator_class.get_class_tag("algorithm_type")
+
+    if algorithm_type is None:
```
Comment (TonyBagnall) [inline aeon/testing/estimator_checking/_yield_classification_checks.py:151]: better to just pass the test if algorithm_type is none that pytest skip

**Review by TonyBagnall — CHANGES_REQUESTED** [review]
thanks

**Review by TonyBagnall — APPROVED** [review]
LGTM, thanks

**Pull request merged**